src/main/java/org/opensearch/testcontainers/OpensearchContainer.java

```diff
@@ -0,0 +1,172 @@
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.containers.wait.strategy.WaitStrategy;
+import org.testcontainers.utility.Base58;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * The Opensearch Docker container (single node cluster) which exposes by default ports 9200
+ * (http/https) and 9300 (tcp, deprecated).
+ */
+public class OpensearchContainer extends GenericContainer<OpensearchContainer> {
+    // Default username to connect to Opensearch instance
+    private static final String DEFAULT_USER = "admin";
+    // Default password to connect to Opensearch instance
+    private static final String DEFAULT_PASSWORD = "admin";
+
+    // Default HTTP port.
+    private static final int DEFAULT_HTTP_PORT = 9200;
+
+    // Default TCP port (deprecated and may be removed in future versions).
+    @Deprecated
+    private static final int DEFAULT_TCP_PORT = 9300;
+
+    // Opensearch Docker base image.
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("opensearchproject/opensearch");
+
+    // Use HTTP or HTTPs endpoint
+    private final boolean useHttps;
+
+    /**
+     * Create an Opensearch Container by passing the full docker image name.
+     *
+     * @param dockerImageName Full docker image name as a {@link String}, like:
+     *     opensearchproject/opensearch:1.2.4 opensearchproject/opensearch:1.3.1
+     *     opensearchproject/opensearch:2.0.0
+     */
+    public OpensearchContainer(String dockerImageName) {
+        this(DockerImageName.parse(dockerImageName));
+    }
+
+    /**
+     * Create an Opensearch Container (with security plugin enabled) by passing the full docker image
+     * name.
+     *
+     * @param dockerImageName Full docker image name as a {@link DockerImageName}, like:
+     *
+     *      DockerImageName.parse("opensearchproject/opensearch:1.2.4")
+     *      DockerImageName.parse("opensearchproject/opensearch:1.3.1")
+     *      DockerImageName.parse("opensearchproject/opensearch:2.0.0")
+     *
+     */
+    public OpensearchContainer(final DockerImageName dockerImageName) {
+        this(dockerImageName, false);
+    }
+
+    /**
+     * Create an Opensearch Container by passing the full docker image name.
+     *
+     * @param dockerImageName Full docker image name as a {@link DockerImageName}, like:
+     *
+     *    DockerImageName.parse("opensearchproject/opensearch:1.2.4")
+     *    DockerImageName.parse("opensearchproject/opensearch:1.3.1")
+     *    DockerImageName.parse("opensearchproject/opensearch:2.0.0")
+     *
+     * @param disableSecurity Should the security plugin be enabled or disabled. If the security
+     *     plugin is enabled, HTTPS protocol is going to be used along with the default username /
+     *     password.
+     */
+    public OpensearchContainer(final DockerImageName dockerImageName, boolean disableSecurity) {
```

Comment:
Don't you think it is better to not have this additional public constructor and handle the security simply by using a setter like `withSecurityEnabled()` and a default field value?

I think this is more idiomatic and similar to how other containers handle it.

Reply:
I see your point, the initialization may change a bit since security is impacting the `HttpWaitStrategy` (SSL / plain), I will explore this part

Reply:
@kiview Looked into that, I think the property won't be convenient here for 2 reasons:
 - it impacts container initialization (on/off security)
 - it  impacts `HttpWaitStrategy`

In general, we could move some of the initialization logic into the setter for the price of throwing `IllegalStateException` if container is already created or/and running. The usability of the API would certainly suffer.

Reply:
The code below can be moved to `configure()`

https://github.com/opensearch-project/opensearch-testcontainers/pull/2/files#diff-23a1cc1ef42d4ce0e7691ff930c3046cc7a19b6b4cb93c85f1c4ea260d2dafc4R85-R112

See [this](https://github.com/testcontainers/testcontainers-java/blob/de1324ed2800eff4da326d0c23d281399d006bc0/modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java#L241-L320) example

Reply:
@eddumelendez is right and I would consider this the Testcontainers-idiomatic way to do it 👍

Reply:
@eddumelendez @kiview absolutely, no objections, thank you guys